dependencies/windows/install-boost/install-boost.R

```diff
@@ -1,63 +1,119 @@
-# archive. Even though the '.zip' is the largest, we use it because it works.
-boost_url <- "https://sourceforge.net/projects/boost/files/boost/1.50.0/boost_1_50_0.zip"
-output_file <- "boost-1.50-win-rtools33-gcc493.zip"
-toolchain_bin   <- normalizePath("C:/Rtools/bin", mustWork = TRUE)
-toolchain_32bit <- normalizePath("C:/Rtools/mingw_32/bin", mustWork = TRUE)
-toolchain_64bit <- normalizePath("C:/Rtools/mingw_64/bin", mustWork = TRUE)
+# initialize log directory (for when things go wrong)
+unlink("logs", recursive = TRUE)
+dir.create("logs")
+options(log.dir = normalizePath("logs"))
 
-# validate paths
-if (!file.exists(toolchain_32bit))
-   fatal("No toolchain at path '%s'", toolchain_32bit)
+# put RStudio tools on PATH
+PATH$prepend("../tools")
 
-if (!file.exists(toolchain_64bit))
-   fatal("No toolchain at path '%s'", toolchain_64bit)
+# initialize variables
+boost_url <- "https://sourceforge.net/projects/boost/files/boost/1.63.0/boost_1_63_0.7z"
+output_file <- "boost-1.63-win-rtools33-gcc493.zip"
+output_dir <- file.path("..", tools::file_path_sans_ext(output_file))
+toolchain_bin   <- normalizePath("../Rtools33/bin", mustWork = TRUE)
+toolchain_32bit <- normalizePath("../Rtools33/mingw_32/bin", mustWork = TRUE)
+toolchain_64bit <- normalizePath("../Rtools33/mingw_64/bin", mustWork = TRUE)
+
+# clear out the directory we'll create boost in
+unlink(output_dir, recursive = TRUE)
+ensure_dir(output_dir)
+output_dir <- normalizePath(output_dir)
+
+# boost modules we need to alias
+boost_modules <- c(
+   "algorithm", "asio", "array", "bind", "chrono", "circular_buffer",
+   "context", "crc", "date_time", "filesystem", "foreach", "format",
+   "function", "interprocess", "iostreams", "lambda", "lexical_cast",
+   "optional", "program_options", "random", "range", "ref", "regex",
+   "scope_exit", "signals", "smart_ptr", "spirit", "string_algo",
+   "system", "test", "thread", "tokenizer", "type_traits", "typeof",
+   "unordered", "utility", "variant"
+)
 
 # construct paths of interest
 boost_filename <- basename(boost_url)
 boost_dirname <- tools::file_path_sans_ext(boost_filename)
-boost_zippath <- boost_filename
+boost_archive <- boost_filename
 
 if (!file.exists(boost_dirname)) {
    
-   section("Downloading Boost...")
-   
-   download(boost_url, destfile = boost_zippath)
-   if (!file.exists(boost_zippath))
-      fatal("Failed to download '%s'", boost_zippath)
+   # download boost if we don't have it
+   if (!file.exists(boost_archive)) {
+      section("Downloading Boost...")
+      download(boost_url, destfile = boost_archive)
+      if (!file.exists(boost_archive))
+         fatal("Failed to download '%s'", boost_archive)
+   }
    
    # extract boost (be patient, this will take a while)
    progress("Extracting archive -- please wait a moment...")
-   exec("unzip", boost_filename)
+   progress("Feel free to get up and grab a coffee.")
+   exec("7z.exe", "x -mmt4", boost_filename)
```

Comment:
I use `7z.exe` here because both of our `zip.exe` and `tar.exe` utilities actually barf when extracting the Boost sources, as the archives contain some non-portably long paths (for documentation files).